storage/helpers/customDatabaseRemover.go

```diff
@@ -0,0 +1,95 @@
+package helpers
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-go/config"
+)
+
+const (
+	patternItemsSeparator = ","
+	divisibleByChar       = "%"
+)
+
+type customDatabaseRemover struct {
+	shouldRemoveHandler func(dbIdentifier string, epoch uint32) bool
+}
+
+// NewCustomDatabaseRemover returns a new instance of customDatabaseRemover
+func NewCustomDatabaseRemover(cfg config.StoragePruningConfig) (*customDatabaseRemover, error) {
+	if len(cfg.AccountsTrieSkipRemovalCustomPattern) == 0 {
+		return &customDatabaseRemover{
+			shouldRemoveHandler: func(_ string, _ uint32) bool {
+				return true
+			},
+		}, nil
+	}
+
+	shouldRemoveFunc, err := parseCustomPattern(cfg.AccountsTrieSkipRemovalCustomPattern)
+	if err != nil {
+		return nil, fmt.Errorf("%w while creating a custom database remover", err)
+	}
+
+	return &customDatabaseRemover{
+		shouldRemoveHandler: shouldRemoveFunc,
+	}, nil
+}
+
+// ShouldRemove returns the result of the inner handler function
+func (c *customDatabaseRemover) ShouldRemove(dbIdentifier string, epoch uint32) bool {
+	return c.shouldRemoveHandler(dbIdentifier, epoch)
+}
+
+func parseCustomPattern(pattern string) (func(dbIdentifier string, epoch uint32) bool, error) {
+	epochs := make([]uint32, 0)
+	splitArgs := strings.Split(pattern, patternItemsSeparator)
+	for _, arg := range splitArgs {
```

Comment:
this should be split and called only in the constructor to create the "divisibleEpochs" variable